rclcpp_action/include/rclcpp_action/client_goal_handle.hpp

```diff
@@ -32,24 +48,72 @@ template<typename ACTION>
 class ClientGoalHandle
 {
 public:
+  RCLCPP_SMART_PTR_DEFINITIONS_NOT_COPYABLE(ClientGoalHandle)
+
+  // A wrapper that defines the result of an action
+  typedef struct Result
+  {
+    /// The unique identifier of the goal
+    GoalID goal_id;
+    /// A status to indicate if the goal was canceled, aborted, or suceeded
+    ResultCode code;
+    /// User defined fields sent back with an action
+    typename ACTION::Result::SharedPtr response;
+  } Result;
+
+  using Feedback = typename ACTION::Feedback;
+  using FeedbackCallback = std::function<void (ClientGoalHandle::SharedPtr, const Feedback &)>;
```

Comment:
Is there a reason that the feedback and result are given as references, but in the server API they are provided as shared pointers (to goals, etc...)?

Reply:
For example: https://github.com/ros2/rclcpp/pull/593/files#diff-83f58bb2e2044a426e1186259c2a409fR282

Reply:
Changed to shared pointer in 923564b76e64acde0f38caacf5a658e56090090d